### PR TITLE
Add quests actions to Habitica

### PR DESCRIPTION
**Pull request by tr4nt0r**
## Proposed change

Adds actions for quest handling in Habitica

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/129650
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Introduced new quest management actions in the Habitica integration:
		- Accept or reject pending quest invitations.
		- Leave an ongoing quest.
		- Abort or start quests as a leader.
		- Cancel quests that have not yet started.
	- Added an automation example for auto-accepting quest invitations, enhancing user experience by streamlining quest participation notifications.

These enhancements provide users with improved control over their quest participation directly from Home Assistant.

### CODE REVIEW

File: source/_integrations/habitica.markdown [hunk 1]
```diff
@@ -130,6 +130,30 @@ Use a skill or spell from your Habitica character on a specific task to affect i
...
+
+Terminate your party's ongoing quest. All progress will be lost and the quest roll returned to the owner's inventory. Only quest leader or group leader can perform this action.
+
+### Action `habitica.start_quest` 🔒
+
+Begin the quest immediately, bypassing any pending invitations that haven't been accepted or rejected. Only quest leader or group leader can perform this action.
+
+### Action `habitica.cancel_quest` 🔒
+
+Cancel a quest that has not yet startet. All accepted and pending invitations will be canceled and the quest roll returned to the owner's inventory. Only quest leader or group leader can perform this action.
```
Comment (c0ffeeca7) [inline source/_integrations/habitica.markdown]:
```suggestion
Cancel a quest that has not yet started. All accepted and pending invitations will be canceled, and the quest roll returned to the owner's inventory. Only the quest leader or group leader can perform this action.
```

File: source/_integrations/habitica.markdown [hunk 2]
```diff
@@ -130,6 +130,30 @@ Use a skill or spell from your Habitica character on a specific task to affect i
...
+
+Leave the current quest you are participating in.
+
+### Action `habitica.abort_quest` 🔒
+
+Terminate your party's ongoing quest. All progress will be lost and the quest roll returned to the owner's inventory. Only quest leader or group leader can perform this action.
+
+### Action `habitica.start_quest` 🔒
+
+Begin the quest immediately, bypassing any pending invitations that haven't been accepted or rejected. Only quest leader or group leader can perform this action.
```
Comment (c0ffeeca7) [inline source/_integrations/habitica.markdown]:
```suggestion
Begin the quest immediately, bypassing any pending invitations that haven't been accepted or rejected. Only the quest leader or group leader can perform this action.
```

File: source/_integrations/habitica.markdown [hunk 3]
```diff
@@ -130,6 +130,30 @@ Use a skill or spell from your Habitica character on a specific task to affect i
...
+
+Reject a pending invitation to a quest.
+
+### Action `habitica.leave_quest`
+
+Leave the current quest you are participating in.
+
+### Action `habitica.abort_quest` 🔒
+
+Terminate your party's ongoing quest. All progress will be lost and the quest roll returned to the owner's inventory. Only quest leader or group leader can perform this action.
```
Comment (c0ffeeca7) [inline source/_integrations/habitica.markdown]:
```suggestion
Terminate your party's ongoing quest. All progress will be lost, and the quest roll returned to the owner's inventory. Only the quest leader or group leader can perform this action.
```

**Review by c0ffeeca7 — DISMISSED** [review]
Thank you, @tr4nt0r :+1:

**Comment by tr4nt0r** [comment]
Added an example automation/blueprint and incorporated some of the coderabbit suggestions. Although not sure about the positioning of the Note box below the actions section. Maybe at the beginning of the actions section or were the restricted actions begin are better positions?

**Comment by c0ffeeca7** [comment]
> Added an example automation/blueprint and incorporated some of the coderabbit suggestions. Although not sure about the positioning of the Note box below the actions section. Maybe at the beginning of the actions section or were the restricted actions begin are better positions?

Nice! :tada: 

Note: Yes, that one is tricky, because people probably don't read top to bottom. 
I agree with you, I would put it right under the first occurrence

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @tr4nt0r :+1:

**Pull request merged**